src/mpid/ch4/netmod/ofi/ofi_types.h

```diff
@@ -397,6 +350,23 @@ typedef struct {
     char kvsname[MPIDI_KVSAPPSTRLEN];
     char pname[MPI_MAX_PROCESSOR_NAME];
     int port_name_tag_mask[MPIR_MAX_CONTEXT_MASK];
+
+    /* Capability settings */
+#ifdef MPIDI_OFI_ENABLE_RUNTIME_CHECKS
```

Comment:
There seems to be some confusion here.  @charlesarcher says that you can pass in a "NULL" input parameter and get the capability sets available from the provider.  When you pass `--with-device=ch4:ofi:foobar`, you are explicitly picking/building a single provider, so you should get a single capability set back from the `fi_check_capabilities` function.  After that you are simply asserting whether what you got back is the same as what we hardcoded for that provider.